File: frontend/src/app/dto/invoice.dto.ts
import { UserBasicDTO } from './userBasic.dto';
import { OrderDTO } from './order.dto';

export interface InvoiceDTO {
  id: number;
  invNo: string;
  user: UserBasicDTO;
  order: OrderDTO;
  taxableBase: number;
  totalPrice: number;
  invoiceDate: string;
}

File: backend/src/main/java/es/codeurjc/quesosbartolome/dto/InvoiceDTO.java
package es.codeurjc.quesosbartolome.dto;

import java.time.LocalDateTime;

public record InvoiceDTO(
        Long id,
        String invNo,
        UserBasicDTO user,
        OrderDTO order,
        Double taxableBase,
        Double totalPrice,
        LocalDateTime invoiceDate) {
}
